soroban-env-host/src/cost_runner/cost_types/im_vec_ops.rs

```diff
@@ -6,30 +6,29 @@ type HostVec = MeteredVector<RawVal>;
 pub struct ImVecNewRun;
 impl CostRunner for ImVecNewRun {
     const COST_TYPE: CostType = CostType::VecNew;
-    type SampleType = ();
+    type SampleType = Vec<RawVal>;
 
-    fn run_iter(host: &crate::Host, _iter: u64, _sample: Self::SampleType) {
-        HostVec::new(host.as_budget()).unwrap();
+    fn run_iter(host: &crate::Host, _iter: u64, sample: Self::SampleType) {
+        // `forget` avoids deallocation of sample which artificially inflates the cost
+        std::mem::forget(HostVec::from_vec(sample, host.as_budget()).unwrap());
     }
 }
 
-pub struct ImVecImmutEntryRun;
+pub struct ImVecEntryRun;
 #[derive(Clone)]
-pub struct ImVecImmutEntrySample {
+pub struct ImVecEntrySample {
     pub vec: HostVec,
     pub idxs: Vec<usize>,
 }
-impl CostRunner for ImVecImmutEntryRun {
+impl CostRunner for ImVecEntryRun {
     const COST_TYPE: CostType = CostType::VecEntry;
-    type SampleType = ImVecImmutEntrySample;
+    type SampleType = ImVecEntrySample;
 
-    fn run_iter(host: &Host, iter: u64, sample: Self::SampleType) {
-        let _ = sample
-            .vec
-            .get(
-                sample.idxs[iter as usize % sample.idxs.len()],
-                host.as_budget(),
-            )
-            .unwrap();
+    fn run_iter(host: &Host, _iter: u64, sample: Self::SampleType) {
+        for i in &sample.idxs {
```

Comment:
In order to emulate random access you probably need to do n times random index. Though not sure how relevant this is if we expect our vectors to be small enough, so that cache misses don't matter.